src/finalFrame.js

```diff
@@ -0,0 +1,69 @@
+function FinalFrame() {
+  this.rolls = [];
+  this.strikes = [false, false];
+  this.spare = false;
+  this.remaining = 10;
+}
+
+FinalFrame.prototype.roll = function roll(score) {
+  if (this.over === true) {
+    throw new Error('The frame is already over.');
+  } else if (score > 10) {
+    throw new Error('The score on one roll cannot be over 10.');
+  } else if (score > this.remaining) {
+    throw new Error('You cannot knock down more pins than there are standing.');
+  }
+  this.scoreRoll(score);
+};
+
+FinalFrame.prototype.scoreRoll = function scoreRoll(score) {
+  this.rolls.push(score);
+  if (this.rolls.length === 3) {
+    this.over = true;
+  } else if (this.rolls.length === 2) {
+    this.secondRoll();
+  } else if (this.rolls.length === 1) {
+    this.firstRoll();
+  } else {
+    throw new Error('Incorrect rolls length.');
+  }
+};
+
+FinalFrame.prototype.firstRoll = function firstRoll() {
```

Comment:
Anywhere where I see a number in the method I am suspicious - could this be refactored? I think you could simplify this but just having a 'roll' function and using array indices. You may feel like there is extra logic on the second roll, but I think that could be generalised.